app/src/main/java/com.sun/ui/base/BaseViewModel.kt

```diff
@@ -25,6 +25,12 @@ abstract class BaseViewModel : ViewModel() {
     // viewModelScope with exception handler
     protected val viewModelScopeExceptionHandler = viewModelScope + exceptionHandler
 
+    protected fun executeTask(block: suspend CoroutineScope.() -> Unit) {
```

Comment:
@trungnt-2399  cái block này nó sẽ return về 1 object data đúng k e nhỉ

Reply:
K nhất thiết phải là object data a, nó có thể return về bất cứ thứ gì. Đồng thời nó là 1 scope nên mình hoàn toàn có thể tạo thêm coroutines con để xử lý task nào đó ạ